eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -26,6 +26,10 @@
     "AdditionalTestArguments": [
         "/p:UseProjectReferenceToAzureClients=false",
         "/p:UseProjectReferenceToAzureClients=true"
+    ],
```

Comment:
We currently handle this at https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/templates/variables/globals.yml#L7 today. Any reason we want to add this here as well?

Reply:
Right now PRs only test Debug configuration and private pipeline only test release. This change makes it so we test both debug and release in PRs.

Reply:
What tests do we have that would fail in one or the other configuration? 

If we do want to add these configurations to the matrix we should probably switch purely to the matrix and remove it from globals.yml.

Reply:
Debug.Assert or `#if DEBUG` in product code. Management pipelines use that logic from globals.yml

Reply:
Yeah and we also use the globals.yml configurations for our other jobs like build. The matrix is only used for testing.